### PR TITLE
New Port: Biolab Disaster

### PR DESCRIPTION
**Pull request by scemino**
# New Port for Biolab Disaster

## Game Information
- **Title**: Biolab Disaster
- **URL**: https://github.com/scemino/z_biolab

## Notes

Thanks to [Dominic Szablewski](https://github.com/phoboslab/) for creating this awesome game and game engine.

## Controls

| Button | Action |
|--|--| 
|DPAD|Move|
|A|Shoot|
|B|Jump|

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480
- [ ] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

File: ports/biolab/Biolab Disaster.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+get_controls
+
+GAMEDIR="/$directory/ports/biolab"
+cd $GAMEDIR
+
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+$ESUDO chmod 666 /dev/tty1
+
+export DEVICE_ARCH="${DEVICE_ARCH:-aarch64}"
```
Comment (cdeletre) [inline ports/biolab/Biolab Disaster.sh:28]: Not sure we still need this line.

**Comment by cdeletre** [comment]
Only 640x480 has been validated ?

**Comment by scemino** [comment]
> Only 640x480 has been validated ?
> 
> Looking at the testing thread I suppose more resolutions are ok. But you know more than me :)

Sorry I validated 480x320 with my RG351M.

**Comment by cdeletre** [comment]
> ✅ 353VS / OGA Arkos, Ace Rocknix and 35xxSP muOS. The game seems to run fullscreen no matter what, so its a bit stretched on 4:3 and I bet it will look hillarious on 1:1 screens, but plays fine nonetheless

I think you can add high resolution (Ace test) 🙃

File: ports/biolab/port.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "desc": "A jump'n'gun platformer, displaying many of high_impacts capabilities.",
+    "desc_md": null,
+    "inst": "Ready to run.",
+    "inst_md": null,
+    "genres": [
+      "platformer"
+    ],
+    "image": null,
+    "rtr": true,
+    "exp": true,
```
[the remark this answers is not in the document]
Comment (scemino) [reply ports/biolab/port.json:23]: When do you check this? I don't know ☺️

File: ports/biolab/Biolab Disaster.sh (hunk 1 above)
Comment (scemino) [reply ports/biolab/Biolab Disaster.sh:28]: Which line 26 or 28?

Comment (Cebion) [inline ports/biolab/Biolab Disaster.sh:28]: This one is not needed anymore. Was for transitioning in multiarch ports.

File: ports/biolab/biolab/.DS_Store
Comment (Cebion) [inline ports/biolab/biolab/.DS_Store:1]: mac files have to be cleaned

File: ports/biolab/port.json (hunk 2 above)
Comment (Cebion) [inline ports/biolab/port.json:23]: This line, if its experimental ti won'T show up in Portmaster unless you know what you're doing.

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/biolab/biolab/.DS_Store
Comment (scemino) [reply ports/biolab/biolab/.DS_Store:1]: > mac files have to be cleaned

Sorry about that, I'm not familiar with WEB UI. I'm still figuring out how to remove these files from the WEB.

Comment (scemino) [reply ports/biolab/biolab/.DS_Store:1]: That was painful but it's done.

File: ports/biolab/cover.png
Comment (JanTrueno) [inline ports/biolab/cover.png:1]: Cover is wrong resolution. Needs to be 640x480.

File: ports/biolab/screenshot.png
Comment (JanTrueno) [inline ports/biolab/screenshot.png:1]: Also needs to be 640x480 in resolution :)

**Review by JanTrueno — CHANGES_REQUESTED** [review]
Portfiles are not in the correct structure, please reference this:
- portname/
  - port.json
  - README.md
  - screenshot.jpg
  - gameinfo.xml
  - cover.jpg (Optional)
  - Port Name.sh
  - portname/
    - <portfiles here>
    - LICENSE Files

File: ports/biolab/screenshot.png
Comment (scemino) [reply ports/biolab/screenshot.png:1]: sorry I misunderstood this: https://portmaster.games/packaging.html#screenshot

> The screenshot has to be at least 640x480 in dimensions

**Comment by scemino** [comment]
> Portfiles are not in the correct structure, please reference this:
> 
>     * portname/
>       
>       * port.json
>       * README.md
>       * screenshot.jpg
>       * gameinfo.xml
>       * cover.jpg (Optional)
>       * Port Name.sh
>       * portname/
>         
>         * 
>         * LICENSE Files

What file is not in the correct place? I don't see my error.

**Comment by JanTrueno** [comment]
Seems like the instructions on the website arent correct. Your screenshot and cover look good now. About the structure, you have the cover/screenshot/xml/json all in the portfolder. They should be one folder higher, next to the README and script.

**Comment by scemino** [comment]
> Seems like the instructions on the website arent correct. Your screenshot and cover look good now. About the structure, you have the cover/screenshot/xml/json all in the portfolder. They should be one folder higher, next to the README and script.

This seems all good no? They are next to the README file.
Screenshot 2024-09-17 at 13 16 18

**Comment by JanTrueno** [comment]
Interesting, i have a different view on my end. 
Knipsel

**Comment by scemino** [comment]
> Interesting, i have a different view on my end. Knipsel

This is the same thing, the biolab folder is not expanded, so the files are not in the biolab subfolder.

**Review by JanTrueno — APPROVED** [review]
I was having a brainfart. Looks good ^^

**Pull request merged**